pinot-core/src/main/java/org/apache/pinot/core/data/aggregator/SumPrecisionValueAggregator.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.data.aggregator;
+
+import java.math.BigDecimal;
+import org.apache.pinot.common.function.AggregationFunctionType;
+import org.apache.pinot.spi.data.FieldSpec.DataType;
+import org.apache.pinot.spi.utils.BigDecimalUtils;
+
+
+public class SumPrecisionValueAggregator implements ValueAggregator<Object, BigDecimal> {
+  public static final DataType AGGREGATED_VALUE_TYPE = DataType.BYTES;
+
+  private int _maxByteSize;
+
+  @Override
+  public AggregationFunctionType getAggregationType() {
+    return AggregationFunctionType.SUMPRECISION;
+  }
+
+  @Override
+  public DataType getAggregatedValueType() {
+    return AGGREGATED_VALUE_TYPE;
+  }
+
+  @Override
+  public BigDecimal getInitialAggregatedValue(Object rawValue) {
+    BigDecimal initialValue = toBigDecimal(rawValue);
+    _maxByteSize = Math.max(_maxByteSize, BigDecimalUtils.byteSize(initialValue));
+    return initialValue;
+  }
+
+  @Override
+  public BigDecimal applyRawValue(BigDecimal value, Object rawValue) {
+    value = value.add(toBigDecimal(rawValue));
+    _maxByteSize = Math.max(_maxByteSize, BigDecimalUtils.byteSize(value));
+    return value;
+  }
+
+  private static BigDecimal toBigDecimal(Object rawValue) {
+    if (rawValue instanceof byte[]) {
+      return BigDecimalUtils.deserialize((byte[]) rawValue);
+    }
+    if (rawValue instanceof Integer || rawValue instanceof Long) {
+      return BigDecimal.valueOf(((Number) rawValue).longValue());
+    }
+    return new BigDecimal(rawValue.toString());
```

Comment:
Do we need to handle any edge cases and throw ?

Reply:
If the value can be parsed, we parse the value. Otherwise, the constructor should throw the exception